api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -43,11 +58,51 @@
 
   @Override
   public ResponseEntity<Profile> register(RegistrationRequest registrationRequest) {
+    User user = userProvider.get();
+    if (user.getDataAccessLevel() != DataAccessLevel.UNREGISTERED) {
+      throw new BadRequestException("User {0} is already registered".format(user.getEmail()));
+    }
     try {
-      // TODO: do registration, etc.
-      return ResponseEntity.ok(profileService.getProfile());
+      fireCloudService.registerUser(registrationRequest.getContactEmail(),
+          registrationRequest.getGivenName(), registrationRequest.getFamilyName());
     } catch (ApiException e) {
-      log.log(Level.INFO, "Error calling FireCloud", e);
+      log.log(Level.SEVERE, "Error registering user: {0}".format(e.getResponseBody()), e);
+      // We don't expect this to happen.
+      // TODO: figure out what happens if you register after already registering
+      throw new ServerErrorException("Error registering user", e);
+    }
+    String billingProjectName = "all-of-us-free-" + user.getUserId();
```

Comment:
As mentioned over Slack: this doesn't lead to distinct values in different environments. At a minimum we should probably put some environment string in here (e.g. "test", "dev", "prod"). At least for dev purposes, we should probably also be OK with reusing the same billing projects repeatedly for the same or different users, so we don't create a bajillion orphaned billing projects.

I haven't quite figured out what we want here.

Reply:
If user ID is unique, we shouldn't get any name collisions here, right? 

I agree that I think we should have dev/testing accounts that we share so that we use the same billing projects.

Reply:
User ID here is not a UUID, or an ID generated based off an e-mail address, but a database ID integer... it isn't unique across different environments with different DBs.

We could generate some kind of UUID, but I was thinking having a DB based off the user ID would be fine if we had a prefix per environment that we care to distinguish. (We could share a prefix for dev / test.)

I may postpone this until we have the configuration setup I mentioned over Slack.

Reply:
In that case I think the different environment variables makes sense. I'm happy calling that good for now in that case.